lib/puppet/network/http/api/indirected_routes.rb

```diff
@@ -192,7 +192,12 @@ def read_body_into_model(model_class, request)
     data = request.body.to_s
 
     format = request.format
-    model_class.convert_from(format, data)
+    begin
+      model_class.convert_from(format, data)
+    rescue => e
+      raise Puppet::Network::HTTP::Error::HTTPBadRequestError.new(
```

Comment:
Whether 400 vs 422 for the body has been a point of debate for a while - https://stackoverflow.com/questions/16133923/400-vs-422-response-to-post-of-data - currently 400 seems to be the way to go.